enterprise/backend/src/metabase_enterprise/semantic_search/index.clj

```diff
@@ -72,38 +78,80 @@
      :legacy_input        [:cast (json/encode legacy_input) :jsonb]
      :metadata            [:cast (json/encode doc) :jsonb]}))
 
+(defn- batch-update!
+  ([records->sql documents]
+   (jdbc/with-transaction [tx @db/data-source]
```

Comment:
Do we want these updates wrapped in a single transaction? Leaning towards removing it.

Reply:
Seems reasonable to me that we wouldn't need a transaction here